app/src/main/java/edu/outside2154/gamesense/fragment/ChecklistFragment.kt

```diff
@@ -0,0 +1,94 @@
+package edu.outside2154.gamesense.fragment
+
+import android.content.Context
+import android.net.Uri
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+import edu.outside2154.gamesense.R
+
+/**
+ * A simple [Fragment] subclass.
+ * Activities that contain this fragment must implement the
+ * [ChecklistFragment.OnFragmentInteractionListener] interface
+ * to handle interaction events.
+ * Use the [ChecklistFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class ChecklistFragment : Fragment() {
+
+    // TODO: Rename and change types of parameters
+    private var mParam1: String? = null
+    private var mParam2: String? = null
+
+    private var mListener: OnFragmentInteractionListener? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (arguments != null) {
+            mParam1 = arguments.getString(ARG_PARAM1)
+            mParam2 = arguments.getString(ARG_PARAM2)
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
+        return inflater!!.inflate(R.layout.fragment_checklist, container, false)
+    }
+
+    // TODO: Rename method, update argument and hook method into UI event
+    fun onButtonPressed(uri: Uri) {
+        if (mListener != null) {
+            mListener!!.onFragmentInteraction(uri)
```

Comment:
No need for unsafe cast, `mListener` has been smart-casted into a non-nullable type due to the `mListener != null` check. In fact, we should rewrite the entire function to:

```kotlin
fun onButtonPressed(uri: Uri) = mListener?.onFragmentInteraction(uri)
```